Source/playerdat.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file playerdat.hpp
+ *
+ * Interface of all player data.
+ */
+#pragma once
+
+#include <cstdint>
+
+#include "player.h"
+#include "textdat.h"
+
+namespace devilution {
+
+struct PlayerData {
+	/* Class Name */
+	const char *className;
+	/* Class Directory Path */
+	const char *classPath;
+	/* Class Starting Strength Stat */
+	uint8_t baseStr;
+	/* Class Starting Magic Stat */
+	uint8_t baseMag;
+	/* Class Starting Dexterity Stat */
+	uint8_t baseDex;
+	/* Class Starting Vitality Stat */
+	uint8_t baseVit;
+	/* Class Maximum Strength Stat */
+	uint8_t maxStr;
+	/* Class Maximum Magic Stat */
+	uint8_t maxMag;
+	/* Class Maximum Dexterity Stat */
+	uint8_t maxDex;
+	/* Class Maximum Vitality Stat */
+	uint8_t maxVit;
+	/* Class Block Bonus % */
+	uint8_t blockBonus;
+	/* Life gained on level up */
+	uint16_t lvlUpLife;
+	/* Mana gained on level up */
+	uint16_t lvlUpMana;
+	/* Life from base Vitality */
+	uint16_t chrLife;
+	/* Mana from base Magic */
+	uint16_t chrMana;
+	/* Life from item bonus Vitality */
+	uint16_t itmLife;
+	/* Mana from item bonus Magic */
+	uint16_t itmMana;
+	/* Nothing (Idle) frame count */
+	int8_t NFrames;
+	/* Walking frame count */
+	int8_t WFrames;
+	/* Blocking frame count */
+	int8_t BFrames;
+	/* Death frame count */
+	int8_t DFrames;
+	/* Spellcasting frame count */
+	int8_t SFrames;
+	/* Hit Recovery frame count */
+	int8_t HFrames;
+	/* Town Nothing (Idle) frame count */
+	int8_t TNFrames;
+	/* Town Walking frame count */
+	int8_t TWFrames;
+	/* Spellcasting action frame number */
+	int8_t SFNum;
+	/* Class Skill */
+	SpellID skill;
+	/* Sprite width: Stand */
+	uint8_t swStand;
+	/* Sprite width: Walk */
+	uint8_t swWalk;
+	/* Sprite width: Attack */
+	uint8_t swAttack;
+	/* Sprite width: Attack (Bow) */
+	uint8_t swBow;
+	/* Sprite width: Hit Recovery */
+	uint8_t swHit;
+	/* Sprite width: Block */
+	uint8_t swBlock;
+	/* Sprite width: Cast Lightning Spell */
+	uint8_t swLightning;
+	/* Sprite width: Cast Fire Spell */
+	uint8_t swFire;
+	/* Sprite width: Cast Magic Spell */
+	uint8_t swMagic;
+	/* Sprite width: Death */
+	uint8_t swDeath;
+};
+
+struct PlayerAttackAnimData {
+	/* Unarmed frame count */
+	int8_t AFramesUnarmed;

```

Comment:
The fields should be named in `camelCase`, for example `framesUnarmed`.
If you use VS Code, it should be complaining about this btw.

Reply:
Done